pandas/sparse/array.py

```diff
@@ -563,7 +572,9 @@ def make_sparse(arr, kind='block', fill_value=nan):
     """
 
     arr = _sanitize_values(arr)
-    length = len(arr)
+
+    if np.ndim(arr) > 1:
+        raise TypeError("expected dimension <= 1 data")
```

Comment:
Yes.  Look at my changes in `sparse/tests/test_array`.  I'll create a separate PR for this as you mentioned in the original issue.